subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -180,3 +182,20 @@ int zsock_getaddrinfo(const char *host, const char *service,
 	}
 	return ret;
 }
+
+#else /* CONFIG_DNS_RESOLVER */
+
+/* For kernel/userspace mode with syscalls, we always must have getaddrinfo
```

Comment:
This comment isn't correct.
The syscall mechanism gracefully handles system calls where the implementation is missing and reports an error at runtime.
Strictly speaking, you could elide this empty function body if you wanted to, it's not a requirement for system calls.

Reply:
> The problem we are solving here is that if getaddrinfo() implementation is built and not removed from the binary via gc-sections, that we don't get linker errors if DNS isn't also compiled.

Then I indeed misunderstand what the problem we solve. Because in https://github.com/zephyrproject-rtos/zephyr/pull/9038#issuecomment-406384073, you show that we **do** get a linker error if getaddrinfo() is called, and DNS resolver isn't enabled.

Overall, please suggest what should be written in that comment/if it should be removed.